tests/client/client_tests.py

```diff
@@ -81,6 +81,13 @@ def test_system_info(elasticapm_client):
     assert {"hostname", "architecture", "platform"} == set(system_info.keys())
 
 
+@pytest.mark.parametrize("elasticapm_client", [{"hostname": "my_custom_hostname"}], indirect=True)
```

Comment:
Does it make sense to add a test that checks `hostname` but without an explicitly configured hostname? To make sure that we're populating it either way? (This may be covered by another test, I haven't checked.)

Reply:
Good idea, added (somewhat sub optimally, see commit message) in 5b1b81e0